plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/LibvirtStorageAdaptor.java

```diff
@@ -1609,7 +1609,9 @@ to support snapshots(backuped) as qcow2 files. */
                     } else {
                         destFile = new QemuImgFile(destPath, destFormat);
                         try {
-                            qemu.convert(srcFile, destFile, null, null, new QemuImageOptions(srcFile.getFormat(), srcFile.getFileName(), null), null, false, true);
+                            boolean isQCOW2 = !sourcePath.endsWith(".raw");
```

Comment:
```suggestion
                            boolean isQCOW2 = PhysicalDiskFormat.QCOW2.equals(sourceFormat);
```
I think this is a bit better

Reply:
thanks @JoaoJandre 

verified ok